src/Admin/SnapshotAdmin.php

```diff
@@ -73,9 +78,23 @@ protected function alterNewInstance(object $object): void
     protected function prePersist(object $object): void
     {
         $page = $this->isChild() ? $this->getParent()->getSubject() : $object->getPage();
-        \assert($page instanceof PageInterface);
 
-        $this->transformer->create($page, $object);
+        if (null !== $page) {
+            \assert($page instanceof PageInterface);
+            $this->transformer->create($page, $object);
+        }
+    }
+
+    protected function postPersist(object $object): void
+    {
+        $page = $object->getPage();
+
+        if (null !== $page) {
+            $page->setEdited(false);
+            $this->pageManager->save($page);
+        }
+
+        $this->snapshotManager->enableSnapshots([$object]);
```

Comment:
Basically here we are splitting the https://github.com/sonata-project/SonataPageBundle/blob/4.x/src/Service/CreateSnapshotService.php#L63-L74

in two pieces. First we create the snapshot in pre persist, and once is persisted, we set page edited to false and enable snapshots (to enable snapshot we need the id to be set, that's why we separate in 2 steps).